### ISSUE TITLE
bug: no-construct-in-public-property-of-construct is not working at array type

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#218]
### Description
The `no-construct-in-public-property-of-construct` ESLint rule is not detecting construct types when they are used in an array within public properties.

### Example Code
```typescript
import { Bucket } from "aws-cdk-lib/aws-s3";
import { Construct } from "constructs";

export class MyConstruct extends Construct {
  public readonly buckets: Bucket[]; // not raising an error
  
  constructor(scope: Construct, id: string) {
    super(scope, id);
    this.buckets = [];
  }
}
```

### Expected Behavior
The rule should detect and report an error when CDK Construct types are used in arrays within public properties, similar to how it works for non-array types.

### Actual Behavior
No error is reported when the construct type is used as an array element type.

### Environment
- eslint-cdk-plugin Version: 3.4.0
- Node.js Version: 22.15.0
- Language: TypeScript

### Related Issue
This is the same type of bug as #216 but affects the `no-construct-in-public-property-of-construct` rule instead of `no-construct-in-interface`.
